### PR TITLE
polymorphic data loader in contrib

### PR DESCRIPTION
**Pull request by TonyBagnall**
moved the polymorphic data loader to contrib due to reviewer objections to its presence in main

### CODE REVIEW

File: sktime/contrib/data_io.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            num_cases = num_cases + 1
+            line = line.replace("?", replace_missing_vals_with)
+            dimensions = line.split(":")
+            # If first instance then note the number of dimensions.
+            # This must be the same for all cases.
+            if is_first_case:
+                num_dimensions = len(dimensions)
+                if _meta_data["has_class_labels"]:
+                    num_dimensions -= 1
+                for _dim in range(0, num_dimensions):
```
Comment (chrisholder) [inline sktime/contrib/data_io.py:135]: Remove for loop here just do:

instance_list = [ [] for _ in range(num_dimensions) ]

File: sktime/contrib/data_io.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    is_first_case = True
+    instance_list = []
+    class_val_list = []
+    line_num = 0
+    num_dimensions = 0
+    num_cases = 0
+    # equal_length = True
+    with open(full_file_path_and_name, "r", encoding="utf-8") as file:
+        _meta_data = _read_header(file, full_file_path_and_name)
+        for line in file:  # Will this work?
```
Comment (chrisholder) [inline sktime/contrib/data_io.py:125]: Remove comment

File: sktime/contrib/data_io.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            num_cases = num_cases + 1
```
Comment (chrisholder) [inline sktime/contrib/data_io.py:126]: num_cases += 1

**Review by chrisholder — CHANGES_REQUESTED** [review]

File: sktime/contrib/data_io.py (hunk 1 above)
Comment (TonyBagnall) [reply sktime/contrib/data_io.py:135]: does it make any difference? this is in contrib, its the wild west :) Still, have changed it

**Pull request merged**